packages/vertexai/src/models/imagen-model.test.ts

```diff
@@ -0,0 +1,246 @@
+    await imagenModel.generateImages(prompt, { numberOfImages: 1 });
+    expect(makeRequestStub).to.be.calledWith(
+      'publishers/google/models/my-model',
+      request.Task.PREDICT,
+      match.any,
+      false,
+      match((value: string) => {
+        return (
+          value.includes('safetyFilterLevel') &&
+          value.includes('block_only_high') &&
+          value.includes('personFilterLevel') &&
+          value.includes('allow_adult') &&
+          value.includes('mimeType') &&
+          value.includes('image/jpeg') &&
+          value.includes('addWatermark') &&
+          value.includes('true')
+        );
+      }),
+      undefined
+    );
+    restore();
+  });
+  it('generateImages makes a request to predict with request-level image configs', async () => {
+    const imagenModel = new ImagenModel(fakeVertexAI, {
+      model: 'my-model'
+    });
+
+    const mockResponse = getMockResponse(
+      'unary-success-generate-images-base64.json'
+    );
+    const makeRequestStub = stub(request, 'makeRequest').resolves(
+      mockResponse as Response
+    );
+    const prompt = 'A photorealistic image of a toy boat at sea.';
+    await imagenModel.generateImages(prompt, {
+      numberOfImages: 4,
+      aspectRatio: ImagenAspectRatio.WIDESCREEN,
+      negativePrompt: 'do not hallucinate'
+    });
+    expect(makeRequestStub).to.be.calledWith(
+      'publishers/google/models/my-model',
+      request.Task.PREDICT,
+      match.any,
+      false,
+      match((value: string) => {
+        return (
+          value.includes('sampleCount') &&
+          value.includes('4') &&
+          value.includes('aspectRatio') &&
+          value.includes('16:9') &&
+          value.includes('negativePrompt') &&
+          value.includes('do not hallucinate')
+        );
+      }),
+      undefined
+    );
+
+    restore();
+  });
+  it('throws if prompt blocked', async () => {
+    const mockResponse = getMockResponse(
+      'unary-failure-generate-images-prompt-blocked.json'
+    );
+
+    stub(globalThis, 'fetch').resolves({
+      ok: false,
+      status: 400, // TODO (dlarocque): Get this from the mock response
+      statusText: 'Bad Request',
+      json: mockResponse.json
+    } as Response);
+
+    const imagenModel = new ImagenModel(fakeVertexAI, {
+      model: 'my-model'
+    });
+    try {
+      await imagenModel.generateImages(
+        'A photorealistic image of a toy boat at sea.',
+        { numberOfImages: 1 }
+      );
+    } catch (e) {
+      expect((e as VertexAIError).code).to.equal(VertexAIErrorCode.FETCH_ERROR);
+      expect((e as VertexAIError).message).to.include('400');
+      expect((e as VertexAIError).message).to.include(
+        "Image generation failed with the following error: The prompt could not be submitted. This prompt contains sensitive words that violate Google's Responsible AI practices. Try rephrasing the prompt. If you think this was an error, send feedback. Support codes: 42876398"
```

Comment:
Maybe regex this check in case they change the support code or something. Or I guess it's "to.include" so you can just remove the support code.

Reply:
Good idea. I removed the support code